CityPuzzle/Main Game/QuestPage.xaml.cs

```diff
@@ -13,239 +13,175 @@ namespace CityPuzzle
+            InitializeComponent();
+
+            GameLogic gamelogic = new GameLogic();
+            gamelogic.OnNoQuestsLoaded += NoQuestsLoaded;
+            gamelogic.OnNoLocationFound += CurrentLocationError;
+            gamelogic.OnNoNearbyQuest += NoNearbyQuest;
+            gamelogic.OnQuestStart += QuestStart;
+            gamelogic.OnMaskHide += HideMask;
+            gamelogic.OnQuestCompleted += QuestCompleted;
+
+            gamelogic.StartGame();
         }
 
-        public QuestPage()
+        //========================================= Event subscriber methods ===============================================
+        // Send user to create new puzzle if none where loaded
+        private void NoQuestsLoaded(object sender, EventArgs e)
         {
-            InitializeComponent();
-            Target =Sql.ReadPuzzles();
-            if (Target.Count == 0)
-            {
-                Navigation.PushAsync(new AddObjectPage());
-            }
-            else
-            {
-                ShowQuest();
-            }
+            Navigation.PushAsync(new AddObjectPage());
         }
 
-        async private void ShowQuest()
+        // Alert message when current location was not found
+        private async void CurrentLocationError(object sender, EventArgs e)
         {
-            await UpdateCurrentLocation();
-            Puzzle target = GetQuest(Target);
+            await DisplayAlert("Error", "Nepavyksta aptikti jusu buvimo vietos.", "OK");
+        }
 
-            if (target == null)      // when no nearby quests are found. Suggest creating a new one and exit to meniu
-            {
-                await DisplayAlert("No destinations in " + App.CurrentUser.MaxQuestDistance + " km radius", "Consider creating a nearby destination yourself.", "OK");
-                await Navigation.PopAsync();
-            }
-            else
-            {
-                QuestLat = target.Latitude;
-                QuestLng = target.Longitude;
-                QuestInProgress = target;
-                var questImg = target.ImgAdress;
-                objimg.Source = questImg;
-
-                MissionLabel.Text = "Tavo uzduotis- surasti mane!";
-                QuestField.Text = target.Quest;
-
-                await RevealImg();    // Start the quest completion loop
-                App.CurrentUser.QuestsCompleted.Add(new Lazy<Puzzle>(() =>target));            // TO DO: save user data to database after finishing quest or loging out
-                await DisplayAlert("Congratulations", "You have reached the destination", "OK");
-            }
+        // Show alert message when no quests were found around user and go back to previous window
+        private async void NoNearbyQuest(object sender, EventArgs e)
+        {
+            await DisplayAlert("No destinations in " + App.CurrentUser.MaxQuestDistance + " km radius", "Consider creating a nearby destination yourself.", "OK");
+            await Navigation.PopAsync();
```

Comment:
If no quests were found go back to meniu and not just one page back

Reply:
This is being tested and modified to do exactly thay but it still needs some time